Spigot-Server-Patches/0312-Slime-pathfinder-additions.patch

```diff
@@ -0,0 +1,123 @@
+From a106042312bb31c6f5a7b028801c71c4da6d1529 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Mon, 18 Jun 2018 06:13:12 -0500
+Subject: [PATCH] Slime pathfinder additions
+
+
+diff --git a/src/main/java/net/minecraft/server/EntitySlime.java b/src/main/java/net/minecraft/server/EntitySlime.java
+index 3d3a9ca04..620c60148 100644
+--- a/src/main/java/net/minecraft/server/EntitySlime.java
++++ b/src/main/java/net/minecraft/server/EntitySlime.java
+@@ -2,6 +2,9 @@ package net.minecraft.server;
+ 
+ import javax.annotation.Nullable;
+ // CraftBukkit start
++import com.destroystokyo.paper.event.entity.EntityPathfindEvent;
++import org.bukkit.entity.Slime;
++import org.bukkit.event.entity.SlimeJumpEvent;
+ import org.bukkit.event.entity.SlimeSplitEvent;
+ // CraftBukkit end
+ 
+@@ -306,7 +309,7 @@ public class EntitySlime extends EntityInsentient implements IMonster {
+         }
+ 
+         public boolean a() {
+-            return true;
++            return this.a.canWander && new EntityPathfindEvent(this.a.bukkitEntity, null, null).callEvent(); // Paper - Jumping around
+         }
+ 
+         public void e() {
+@@ -325,12 +328,12 @@ public class EntitySlime extends EntityInsentient implements IMonster {
+         }
+ 
+         public boolean a() {
+-            return this.a.isInWater() || this.a.au();
```

Comment:
This event I don't believe expects null locations... Can you rework this to include locations?